packages/svelte/src/compiler/phases/1-parse/read/context.js

```diff
@@ -121,7 +121,10 @@ function read_type_annotation(parser, optional_allowed = false) {
 	const template =
 		parser.template.slice(0, a).replace(/[^\n]/g, ' ') +
 		insert +
-		parser.template.slice(parser.index);
+		// If this is a type annotation for a function parameter, Acorn-TS will treat subsequent
+		// parameters as part of a sequence expression instead, and will then error on optional
+		// parameters (`?:`). Therefore replace that sequence with something that will not error.
+		parser.template.slice(parser.index).replace(/\?\s*:/g, ':');
```

Comment:
I think this should be safe, this combination is only relevant syntax-wise for optional parameters, there's nothing that should get invalid anywhere else with regards Acorn-TS parsing this - or am I overlooking something?

Reply:
While it's pretty odd to break things up in the following way, it doesn't support comments:

```ts
const f = (n? // foo
  : number): number => n ?? 0 + 1

const g = (n? /* foo */ : number): number => n ?? 0 + 1
```

Those are both valid in normal TypeScript, but broken with this solution.

It also still supports invalid parameter orderings, such as the first being optional and the second being required. Not sure if that's a separate issue or one that this aims to fix (if it even is an issue - it might just be the repl not caring).

Reply:
Yeah this will need to support comments.

Reply:
Looking into both the points you made:

- Supporting comments here would be rare as this is within the template itself – where no one adds comments.
- The invalid ordering of params isn't such an issue here I don't think. This is something for language tools maybe rather than the parser.